src/tests/suite/dfs_unit_test.c

```diff
@@ -0,0 +1,294 @@
+	assert_int_equal(rc, 0);
+	print_message("Container destroying\n");
+	rc = daos_cont_destroy(arg->pool.poh, cuuid, 1, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Destroyed POSIX Container "DF_UUIDF"\n",
+		      DP_UUID(cuuid));
+
+	/** create a DFS container with POSIX layout */
+	rc = dfs_cont_create(arg->pool.poh, cuuid, NULL, NULL, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Created POSIX Container "DF_UUIDF"\n", DP_UUID(cuuid));
+	rc = daos_cont_open(arg->pool.poh, cuuid, DAOS_COO_RW,
+			    &coh, &co_info, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Mounting wrong parameters\n");
+	rc = dfs_mount(arg->pool.poh, coh, -1, &dfs);
+	assert_int_equal(rc, EINVAL);
+	print_message("Container closing\n");
+	rc = daos_cont_close(coh, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Container destroying\n");
+	rc = daos_cont_destroy(arg->pool.poh, cuuid, 1, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Destroyed POSIX Container "DF_UUIDF"\n",
+		      DP_UUID(cuuid));
+
+        /** create a DFS container with POSIX layout */
+	rc = dfs_cont_create(arg->pool.poh, cuuid, NULL, NULL, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Created POSIX Container "DF_UUIDF"\n", DP_UUID(cuuid));
+	rc = daos_cont_open(arg->pool.poh, cuuid, DAOS_COO_RW,
+			    &coh, &co_info, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Mounting NULL dfs\n");
+	rc = dfs_mount(arg->pool.poh, coh, O_RDWR, /*&dfs*/ NULL);
+	assert_int_equal(rc, EINVAL);
+	print_message("Unmount NULL dfs\n");
+	rc = dfs_umount(/*dfs*/ NULL);
+	assert_int_equal(rc, EINVAL);
+	print_message("Container closing\n");
+	rc = daos_cont_close(coh, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Container destroying\n");
+	rc = daos_cont_destroy(arg->pool.poh, cuuid, 1, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Destroyed POSIX Container "DF_UUIDF"\n",
+		      DP_UUID(cuuid));
+}
+
+static void
+dfs_test_open_release(void **state)
+{
+        dfs_obj_t               *obj;
+        daos_size_t             chunk_size = 64;
+        int                     rc;
+
+        rc = dfs_open(/*dfs_mt*/NULL, NULL, "test",
+                      S_IFREG | S_IWUSR | S_IRUSR , O_RDWR | O_CREAT,
+                      OC_S1, chunk_size, NULL, &obj);
+        assert_int_equal(rc, EINVAL);
+        print_message("Mounted file system should be provided\n");
+
+        rc = dfs_open(dfs_mt, NULL, /*"test"*/NULL,
+                      S_IFREG | S_IWUSR | S_IRUSR , O_RDWR | O_CREAT,
+                      OC_S1, chunk_size, NULL, &obj);
+        assert_int_equal(rc, EINVAL);
+        print_message("File name should be provided\n");
+
+        rc = dfs_open(dfs_mt, NULL, "test",
+                      /*S_IFREG | S_IWUSR | S_IRUSR*/ 0, O_RDWR | O_CREAT,
+                      OC_S1, chunk_size, NULL, &obj);
+        assert_int_equal(rc, EINVAL);
+        print_message("mode_t (permissions + type) should be provided\n");
+
+        rc = dfs_open(dfs_mt, NULL, "test",
```

Comment:
this is a valid case. it just fails because the file doesn't exist, hence you get ENOENT and not EINVAL